aggregator/src/aggregator/queue.rs

```diff
@@ -863,4 +1046,91 @@ mod tests {
         install_test_trace_subscriber();
         quickcheck(qc as fn(Parameters) -> TestResult);
     }
+
+    #[test]
+    fn test_request_timeout() {
+        install_test_trace_subscriber();
+
+        tokio::runtime::Runtime::new().unwrap().block_on(async {
+            let meter_prefix = "test";
+            let metrics = InMemoryMetricsInfrastructure::new();
+            let concurrency = 1;
+            let depth = 1;
+            let timeout = Duration::from_millis(100);
+
+            let queue = Arc::new(
+                LIFORequestQueue::new(
+                    concurrency,
+                    depth,
+                    &metrics.meter,
+                    meter_prefix,
+                    Some(timeout),
+                )
+                .unwrap(),
+            );
+
+            // Create a hanging handler that never releases requests
+            let unhang = Arc::new(Notify::new());
+            let handler = Arc::new(queued_lifo(
+                Arc::clone(&queue),
+                HangingHandler {
+                    unhang: Arc::clone(&unhang),
+                },
+            ));
+
+            // Fill up the active slots (concurrency) but leave queue empty
+            let mut requests = Vec::new();
+            for _ in 0..concurrency {
+                let handler = Arc::clone(&handler);
+                requests.push(tokio::spawn(
+                    async move { get("/").run_async(&handler).await },
+                ));
+            }
+
+            // Wait for all active slots to be filled
+            tokio::time::sleep(Duration::from_millis(10)).await;
```

Comment:
when I forward-port this, I want to make this use `wait_for`. I gave up this round.